src/Repository/UserRepository.php

```diff
@@ -108,4 +108,18 @@ public function pagination(string $page): array
 
         return $query->execute();
     }
+
+    public function joinAProgramme(int $userId, Programme $programme): void
+    {
+        $user = $this->findOneBy(['id' => $userId]);
+
+        if (null === $user) {
+            throw new NotFoundHttpException();
```

Comment:
no no no, try to find an exception from doctrine that says not found, not an http one